org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/services/extensions/DocumentLifecycleParticipantTest.java

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.services.extensions;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.eclipse.lemminx.MockXMLLanguageServer;
+import org.eclipse.lemminx.dom.DOMDocument;
+import org.eclipse.lsp4j.TextDocumentContentChangeEvent;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for {@link IDocumentLifecycleParticipant}
+ */
+public class DocumentLifecycleParticipantTest {
```

Comment:
Write tests to make sure that the computed `DOMDocument` is successfully passed to the participant

Reply:
I update test by incrementing the did* only when DOMDocument isnot null (which means that parse has been done). Is thiqkind if texts that you are expecting?

Reply:
Yes, this addresses what I mentioned.